libafl/src/inputs/mod.rs

```diff
@@ -72,17 +86,103 @@ pub trait Input: Clone + Serialize + serde::de::DeserializeOwned + Debug {
         Ok(postcard::from_bytes(&bytes)?)
     }
 
+    /// Serializes this input to the dynamic serialisation format to pass between different fuzzers
+    fn serialize_dynamic(&self, buf: &mut Vec<u8>) -> Result<(), postcard::Error> {
+        buf.extend_from_slice(postcard::to_allocvec(Self::NAME)?.as_slice());
+        buf.extend_from_slice(postcard::to_allocvec(self)?.as_slice());
+        Ok(())
+    }
+
+    /// Deserializes this input type from the dynamic serialization format, if possible
+    fn deserialize_dynamic(
+        buf: &[u8],
+    ) -> Result<Option<Self>, <&mut Deserializer<Slice> as serde::de::Deserializer>::Error> {
+        convert_named(buf)
+    }
+
     /// Generate a name for this input
     fn generate_name(&self, idx: usize) -> String;
 
     /// An hook executed if the input is stored as `Testcase`
     fn wrapped_as_testcase(&mut self) {}
 }
 
+/// Utility trait for downcasting inputs for conversion
+#[cfg(feature = "std")]
```

Comment:
the libs seem to support no_std

Reply:
still inventory doesn't so this stuff is anyway std-only

Reply:
Ah it uses `ctor` I guess. Nevermind

Reply:
So in theory yes, it will work for an ELF with no_std for instance but not for no_std in general. Maybe we should use the ctor feature here to enable it in specific no_std cases.

Reply:
Now you can enable it on no_std with the input_conversion feature, on by default with std

Reply:
Also our registry will check now for ctor and not std, so we can enable ctor on no_std when supported